components/elements/checkbox.vue

```diff
@@ -40,6 +40,12 @@ export default Vue.extend({
     }
   },
 
+  mounted() {
+    if (this.value === true) {
```

Comment:
yes, I like the future proofing aspect of your idea. I will update to:
```ts
if (this.value == true) {
this.checked = !!this.value
}
```

Reply:
Tbh it's a bit of a nitpick but I was more thinking something like this:
```javascript
mounted() {
  this.checked = !!this.value
}
```
which'll make sure this.checked is set to true if a truthy boolean is passed or if a string is passed as value, and will be set to false if a falsy boolean is passed or if no string is passed (at least, if that's the intent). I don't know, one could argue maybe for readability's sake you should keep it as you currently have it hehe.

Reply:
Kevin's approach is ideal. Also, using the not-so-strict comparison with the double equal sign `==` is a bad idea because if you are required to do it it means we are not sure about the value we are comparing. In general, the strict comparison should be applied everywhere, regardless of the type of the primitive